test/common/testlib.py

```diff
@@ -2092,6 +2092,40 @@ def enable_root_login(self):
         if not self.machine.ostree_image and self.file_exists(disallowed_conf):
             self.sed_file('/root/d', disallowed_conf)
 
+    def setup_provisioned_hosts(self, disable_preload=False):
+        """Setup provisioned hosts for testing
+
+        This sets the hostname of all machines to the name given in the
+        provision dictionary and optionally disabled preload when it runs
+        on a development build.
+
+        On Arch Linux add a workaround for systemd/resolvd issue with slow dns
+        lookups.
+        """
+        host_entries = ""
+        for name, m in self.provision.items():
+            print(m)
+            try:
+                address = m['address'].split('/')[0]
+                host_entries += f"{address} {name}\n"
+            except KeyError:
+                continue
+
+        machine = self.machine
+        for name, m in self.machines.items():
+            m.execute(f"hostnamectl set-hostname {name}")
+            # Disable preloading on all machines ("machine1" is done in testlib.py)
+            # Preloading on machines with debug build can overload the browser and cause slowness and browser crashes
+            # In these tests we actually switch between machines in quick succession which can make things even worse
+            self.machine = m
+            if disable_preload and self.is_devel_build():
+                self.disable_preload("packagekit", "playground", "systemd")
+
+            # Workaround systemd DNS issue
+            if machine.image == "arch":
```

Comment:
This will soon hit us on other Fedora /debian-testing etc. images as well, so let's not make it arch specific. Instead, please add a `# HACK` comment pointing to the systemd issue.

Reply:
As Fedora does not use systemd-resolved I think, this should not happen there. This was worked around for Arch in bots by disabling mDNS. We should do the same for the other images.

Reply:
Right, good point. Also, this is obsolete now.